Unmatched rule names of scoped plugins.

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**What version of ESLint are you using?**
- v2.11.1

**What parser (default, Babel-ESLint, etc.) are you using?**
- default.

**Please show your full configuration:**
- I use [@mysticatea/eslint-plugin-test](https://www.npmjs.com/package/@mysticatea/eslint-plugin-test) plugin here.

``` json
{
    "plugins": [
        "@mysticatea/test"
    ],
    "rules": {
        "@mysticatea/test/test": "error"
    }
}
```

`@mysticatea/test/test` rule is:

``` js
    return {
        "Program": function(node) {
            context.report({
                node: node,
                message: "This is a test."
            })
        }
    }
```

**What did you do? Please include the actual source code causing the issue.**
- `eslint test.js`

**What did you expect to happen?**

```
test.js
  1:1  error  This is a test  @mysticatea/test/test

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
test.js
  1:1  error  Definition for rule '@mysticatea/test/test' was not found  @mysticatea/test/test

✖ 1 problem (1 error, 0 warnings)
```

---

I think we should support `@mysticatea/test/test` form's rule specifier.
Thought?

--- a/lib/config/plugins.js
+++ b/lib/config/plugins.js
@@ -75,12 +75,6 @@ module.exports = {
         plugins[shortName] = plugin;
         Environments.importPlugin(plugin, shortName);
         Rules.importPlugin(plugin, shortName);
-
-        // load up environments and rules for the name that '@scope/' was omitted
-        // 3 lines below will be removed by 4.0.0
-        plugins[pluginNameWithoutPrefix] = plugin;
-        Environments.importPlugin(plugin, pluginNameWithoutPrefix);
-        Rules.importPlugin(plugin, pluginNameWithoutPrefix);
     },
 
     /**

--- a/tests/lib/config/plugins.js
+++ b/tests/lib/config/plugins.js
@@ -134,33 +134,33 @@ describe("Plugins", () => {
             assert.equal(Rules.get("@scope/example/foo"), scopedPlugin.rules.foo);
         });
 
-        describe("(NOTE: those behavior will be removed by 4.0.0)", () => {
-            it("should load a scoped plugin when referenced by short name, and should get the plugin even if '@scope/' is omitted", () => {
+        describe("when referencing a scope plugin and omitting @scope/", () => {
+            it("should load a scoped plugin when referenced by short name, but should not get the plugin if '@scope/' is omitted", () => {
                 StubbedPlugins.load("@scope/example");
-                assert.equal(StubbedPlugins.get("example"), scopedPlugin);
+                assert.equal(StubbedPlugins.get("example"), null);
             });
 
-            it("should load a scoped plugin when referenced by long name, and should get the plugin even if '@scope/' is omitted", () => {
+            it("should load a scoped plugin when referenced by long name, but should not get the plugin if '@scope/' is omitted", () => {
                 StubbedPlugins.load("@scope/eslint-plugin-example");
-                assert.equal(StubbedPlugins.get("example"), scopedPlugin);
+                assert.equal(StubbedPlugins.get("example"), null);
             });
 
-            it("should register environments when scoped plugin has environments, and should get the environment even if '@scope/' is omitted", () => {
+            it("should register environments when scoped plugin has environments, but should not get the environment if '@scope/' is omitted", () => {
                 scopedPlugin.environments = {
                     foo: {}
                 };
                 StubbedPlugins.load("@scope/eslint-plugin-example");
 
-                assert.equal(Environments.get("example/foo"), scopedPlugin.environments.foo);
+                assert.equal(Environments.get("example/foo"), null);
             });
 
-            it("should register rules when scoped plugin has rules, and should get the rule even if '@scope/' is omitted", () => {
+            it("should register rules when scoped plugin has rules, but should not get the rule if '@scope/' is omitted", () => {
                 scopedPlugin.rules = {
                     foo: {}
                 };
                 StubbedPlugins.load("@scope/eslint-plugin-example");
 
-                assert.equal(Rules.get("example/foo"), scopedPlugin.rules.foo);
+                assert.equal(Rules.get("example/foo"), null);
             });
         });
     });

